docs/user-guide/mirror_registry_guide.md

```diff
@@ -0,0 +1,263 @@
+  --insecure=$INSECURE
+
+oc image mirror \
+  -a pull_secret.json \
+  quay.io/edge-infrastructure/assisted-installer-controller:latest \
+  $REGISTRY_IP:$REGISTRY_PORT/edge-infrastructure/assisted-installer-controller:latest \
+  --insecure=$INSECURE
+
+oc image mirror \
+  -a pull_secret.json \
+  quay.io/edge-infrastructure/assisted-installer-agent:latest \
+  $REGISTRY_IP:$REGISTRY_PORT/edge-infrastructure/assisted-installer-agent:latest \
+  --insecure=$INSECURE
+```
+
+## Step 2: Apply the CRDs with Updated Configurations
+
+Once the mirror registry is set up, apply the necessary CRDs to configure the cluster to use the mirrored images.
+
+### 1. Apply the Mirror Registry ConfigMap
+
+Create the `mirror_registry.yaml` file as follows:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: mirror-registry-config
+  namespace: <your-namespace>
+data:
+  registries.conf: |
+    [[registry]]
+      prefix = ""
+      location = "quay.io/openshift-release-dev/ocp-release"
+      [[registry.mirror]]
+        location = "$REGISTRY_IP:$REGISTRY_PORT/openshift-release-dev/ocp-release"
+        pull-from-mirror = "digest-only"
+
+    [[registry]]
+      prefix = ""
+      location = "quay.io/edge-infrastructure"
+      [[registry.mirror]]
+        location = "$REGISTRY_IP:$REGISTRY_PORT/edge-infrastructure"
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    ...
+    -----END CERTIFICATE-----
+```
+
+Apply the ConfigMap:
+
+```bash
+oc apply -f mirror_registry.yaml
+```
+
+### 2. Update and Apply CRDs
+
+Ensure that your `AgentClusterInstall` and `InfraEnv` CRDs reference the mirror registry configuration.
+
+**aci.yaml:**
+```yaml
+spec:
+  mirrorRegistryRef:
+    name: mirror-registry-config
+    namespace: <your-namespace>
+```
+
+**infraenv.yaml:**
+```yaml
+spec:
+  mirrorRegistryRef:
+    name: mirror-registry-config
+    namespace: <your-namespace>
+```
+
+Apply the CRDs:
+
+```bash
+oc apply -f aci.yaml
+oc apply -f infraenv.yaml
+```
+
+### 3. Verify the Setup
+
+Ensure that the images are being pulled from the mirror registry and that the cluster is deploying correctly.
+
+## Per-Cluster Mirror Registry Configuration
+
+With this feature, you have the flexibility to define a unique mirror registry for each cluster. This allows you to:
+
+- Specify different registry names and ports for individual clusters.
+- Run multiple mirror registries simultaneously without conflicts by using distinct container names defined by the `REGISTRY_NAME` variable.
+
+For each cluster, simply adjust the `REGISTRY_IP`, `REGISTRY_PORT`, and `REGISTRY_NAME` environment variables to configure a dedicated mirror registry. This setup is particularly useful in multi-cluster environments where isolated registry configurations are required.
+
+## Notes
+
+- **Version Matching:** Ensure that the OpenShift version in the mirror registry matches the version specified in the image set configuration.
```

Comment:
This might be a good note to have under the ```#### 6.1 Mirror OpenShift Release Images``` section above

Reply:
Great idea